providers/implementations/signature/eddsa_sig.c

```diff
@@ -326,6 +509,85 @@ static const OSSL_PARAM *eddsa_gettable_ctx_params(ossl_unused void *vpeddsactx,
     return known_gettable_ctx_params;
 }
 
+static int eddsa_set_ctx_params(void *vpeddsactx, const OSSL_PARAM params[])
+{
+    PROV_EDDSA_CTX *peddsactx = (PROV_EDDSA_CTX *)vpeddsactx;
+    const OSSL_PARAM *p;
+
+    if (peddsactx == NULL)
+        return 0;
+    if (params == NULL)
+        return 1;
+
+    p = OSSL_PARAM_locate_const(params, OSSL_SIGNATURE_PARAM_INSTANCE);
+    if (p != NULL) {
+        char instance_name[OSSL_MAX_NAME_SIZE] = "";
+        char *pinstance_name = instance_name;
+
+        if (!OSSL_PARAM_get_utf8_string(p, &pinstance_name, sizeof(instance_name)))
+            return 0;
+
+        if (OPENSSL_strcasecmp(pinstance_name, SN_Ed25519) == 0) {
```

Comment:
You could have a table for these.. and then just loop over the table in a function.

Reply:
if there were a few more params, then I would do as you suggest.